x/tx/decode/decode.go

```diff
@@ -126,3 +136,56 @@ func (d *Decoder) Decode(txBytes []byte) (*DecodedTx, error) {
 		Signers:                      signers,
 	}, nil
 }
+
+// Hash implements the interface for the Tx interface.
+func (dtx *DecodedTx) Hash() [32]byte {
```

Comment:
These changes were originally in #19588 but I can't convince myself they're needed here since gogotx handles the conversion and tx building.  Is it true they're not needed?

Reply:
Right, if you are not supposed to use a DecodedTx directly, I guess it makes sense to remove this.

Reply:
looks like we should do the reverse: https://github.com/cosmos/cosmos-sdk/pull/20488#discussion_r1622524025

Reply:
Got it, LGTM